core/src/test/scala/com/nvidia/spark/rapids/tool/tuning/BaseAutoTunerSuite.scala

```diff
@@ -68,7 +68,10 @@ class AppInfoProviderMockTest(val maxInput: Double,
  */
 abstract class BaseAutoTunerSuite extends FunSuite with BeforeAndAfterEach with Logging {
 
-  val defaultSparkVersion = "3.1.1"
+  // Default Spark version
+  val defaultSparkVersion = "3.2.0"
```

Comment:
Should we just bump it to the most recent? something like 3.5.x?

Reply:
That makes sense. I have updated the default Spark version in the AutoTuner tests to 3.5.1. T

This also brings up an idea (if needed in future): we could test for both the lower and upper bounds to ensure compatibility across the entire supported range.

Reply:
Thanks @parthosa. That's a good idea.
We can actually read the Spark version from runTime. Doing it dynamically based on the pom will achieve what you just suggested. Then the test will be done for each spark-version in the github workflow.
We can get the spark version using 2 methods:

- From pom file. We can do that by pulling the `build.spark.version`; or doing the easy way
- by getting `ToolUtils.sparkRuntimeVersion`

Reply:
Thanks @amahussein for the suggestion. I think that would be the right thing to do. In that case, we should drop the mvn profiles (<3.2.0). There is a similar issue https://github.com/NVIDIA/spark-rapids/issues/11160 on the plugin side to drop .

Reply:
Summary from offline discussion:
- We should remove the Maven build profiles for Spark versions 3.1.1, 3.1.3, and 3.1.4
- Update the AutoTuner unit tests to dynamically test against multiple Spark versions.